ingestion/src/metadata/ingestion/source/dashboard/superset/metadata.py

```diff
@@ -107,181 +39,6 @@ def create(cls, config_dict: dict, metadata_config: OpenMetadataConnection):
-                    to_entity = self.metadata.get_by_name(
-                        entity=Lineage_Dashboard,
-                        fqn=dashboard_fqn,
-                    )
-                    if from_entity and to_entity:
-                        yield self._get_add_lineage_request(
-                            to_entity=to_entity, from_entity=from_entity
-                        )
-                except Exception as exc:
-                    logger.debug(traceback.format_exc())
-                    logger.error(
-                        f"Error to yield dashboard lineage details for DB service name [{db_service_name}]: {exc}"
-                    )
-
-    def yield_dashboard_chart(
-        self, dashboard_details: dict
-    ) -> Optional[Iterable[CreateChartRequest]]:
-        """
-        Metod to fetch charts linked to dashboard
-        """
-        for chart_id in self._get_charts_of_dashboard(dashboard_details):
-            chart_json = self.all_charts.get(chart_id)
-            if not chart_json:
-                logger.warning(f"chart details for id: {chart_id} not found, skipped")
-                continue
-            chart = CreateChartRequest(
-                name=chart_json["id"],
-                displayName=chart_json.get("slice_name"),
-                description="",
-                chartType=get_standard_chart_type(
-                    chart_json.get("viz_type", ChartType.Other.value)
-                ),
-                chartUrl=chart_json.get("url"),
-                service=EntityReference(
-                    id=self.context.dashboard_service.id.__root__,
-                    type="dashboardService",
-                ),
-            )
-            yield chart
-
-    def _get_datasource_fqn(
-        self, datasource_id: str, db_service_name: str
-    ) -> Optional[str]:
-        if db_service_name:
-            try:
-                datasource_json = self.client.fetch_datasource(datasource_id)
-                database_json = self.client.fetch_database(
-                    datasource_json["result"]["database"]["id"]
-                )
-                dataset_fqn = fqn.build(
-                    self.metadata,
-                    entity_type=Table,
-                    table_name=datasource_json["result"]["table_name"],
-                    schema_name=datasource_json["result"]["schema"],
-                    database_name=database_json["result"]["parameters"]["database"],
-                    service_name=db_service_name,
-                )
-                return dataset_fqn
-            except KeyError as err:
-                logger.debug(traceback.format_exc())
-                logger.warning(
-                    f"Failed to fetch Datasource with id [{datasource_id}]: {err}"
-                )
-        return None
+        if isinstance(connection.connection, ApiConnection):
+            return SupersetAPISource(config, metadata_config)
+        return SupersetDBSource(config, metadata_config)
```

Comment:
I've created a separate source for Database and API, but we don't won't to create a separate config for both of these hence a common source routes API source or DB source depending on the connection type

Reply:
nice, glad we reworked the directories so that we can handle this easily